schemas/stsci.edu/asdf/0.1.0/core/asdf.yaml

```diff
@@ -11,6 +11,20 @@ description: |
 tag: "tag:stsci.edu:asdf/0.1.0/core/asdf"
 type: object
 properties:
+  asdf_library:
+    description: |
+      Describes the ASDF library that produced the file.
+    $ref: software
+
+  software:
+    description: |
+      Lists any relevant software that produced or modified the ASDF
+      file.  May include such things as data calibration pipelines,
+      data analysis tools etc.
+    type: array
+    items:
+      $ref: "software"
```

Comment:
I think if this becomes more like a log, in addition to the software, it each entry should have a free form description along the lines of "applied cosmic ray rejection".